devel/fortio/cmd/fortio/main.go

```diff
@@ -39,46 +39,74 @@ func (f *flagList) Set(value string) error {
-		qpsFlag         = flag.Float64("qps", 8.0, "Queries Per Seconds or 0 for no wait")
-		numThreadsFlag  = flag.Int("c", defaults.NumThreads, "Number of connections/goroutine/threads")
-		durationFlag    = flag.Duration("t", defaults.Duration, "How long to run the test")
-		percentilesFlag = flag.String("p", "50,75,99,99.9", "List of pXX to calculate")
-		resolutionFlag  = flag.Float64("r", defaults.Resolution, "Resolution of the histogram lowest buckets in seconds")
-		compressionFlag = flag.Bool("compression", false, "Enable http compression")
-		goMaxProcsFlag  = flag.Int("gomaxprocs", 0, "Setting for runtime.GOMAXPROCS, <1 doesn't change the default")
-		profileFlag     = flag.String("profile", "", "write .cpu and .mem profiles to file")
-		keepAliveFlag   = flag.Bool("keepalive", true, "Keep connection alive (only for fast http 1.1)")
-		stdClientFlag   = flag.Bool("stdclient", false, "Use the slower net/http standard client (works for TLS)")
-		http10Flag      = flag.Bool("http1.0", false, "Use http1.0 (instead of http 1.1)")
-		grpcFlag        = flag.Bool("grpc", false, "Use GRPC health check")
+var (
+	defaults = &fortio.DefaultRunnerOptions
+	// Very small default so people just trying with random URLs don't affect the target
+	qpsFlag         = flag.Float64("qps", 8.0, "Queries Per Seconds or 0 for no wait")
+	numThreadsFlag  = flag.Int("c", defaults.NumThreads, "Number of connections/goroutine/threads")
+	durationFlag    = flag.Duration("t", defaults.Duration, "How long to run the test")
+	percentilesFlag = flag.String("p", "50,75,99,99.9", "List of pXX to calculate")
+	resolutionFlag  = flag.Float64("r", defaults.Resolution, "Resolution of the histogram lowest buckets in seconds")
+	compressionFlag = flag.Bool("compression", false, "Enable http compression")
+	goMaxProcsFlag  = flag.Int("gomaxprocs", 0, "Setting for runtime.GOMAXPROCS, <1 doesn't change the default")
+	profileFlag     = flag.String("profile", "", "write .cpu and .mem profiles to file")
+	keepAliveFlag   = flag.Bool("keepalive", true, "Keep connection alive (only for fast http 1.1)")
+	stdClientFlag   = flag.Bool("stdclient", false, "Use the slower net/http standard client (works for TLS)")
+	http10Flag      = flag.Bool("http1.0", false, "Use http1.0 (instead of http 1.1)")
+	grpcFlag        = flag.Bool("grpc", false, "Use GRPC (health check) for load testing")
+	echoPortFlag    = flag.Int("http-port", 8080, "http echo server port")
+	grpcPortFlag    = flag.Int("grpc-port", 8079, "grpc port")
+
+	headersFlags flagList
+	percList     []float64
+	err          error
+)
 
-		headersFlags flagList
-	)
+func main() {
 	flag.Var(&headersFlags, "H", "Additional Header(s)")
 	flag.IntVar(&fortio.BufferSizeKb, "httpbufferkb", fortio.BufferSizeKb, "Size of the buffer (max data size) for the optimized http client in kbytes")
 	flag.BoolVar(&fortio.CheckConnectionClosedHeader, "httpccch", fortio.CheckConnectionClosedHeader, "Check for Connection: Close Header")
```

Comment:
oic. well because that's twice the number of lines and doesn't add anything ? those variables are in fortio package and not used directly by the main, I don't think adding more global (even package scope) variables just to move them in a block 3 lines above helps ? also the percentiles has to remain (until cobra) as it's a custom class/parsing